File: go/1701.go
func averageWaitingTime(customers [][]int) float64 {
    total_time := 0
    start := 0
    finish := 0
    for _, customer := range customers{
        first, second := customer[0], customer[1]
        if first > start{
            start = first
        }
        if finish > start{
            start = finish
        }
        finish = start + second
        total_time += finish - first
    }
    
    return float64(total_time)/float64(len(customers))
    
}

File: java/1701.java
class Solution {
    public double averageWaitingTime(int[][] customers) {
        long start = 0;
        long finish = 0;
        long total = 0;
        for(int i = 0; i < customers.length; i ++){
            int a1 = customers[i][0];
            int a2 = customers[i][1];
            
            start = Math.max(finish, a1);
            
            finish = start + a2;
            total += finish - a1;
        }
        return (double) (total) /(double)(customers.length);
        
    }
}
